cmd/operator-sdk/generate/csv.go

```diff
@@ -53,14 +53,9 @@ for the operator. This file is used to publish the operator to the OLM Catalog.
 
 A CSV semantic version is supplied via the --csv-version flag. If your operator
 has already generated a CSV manifest you want to use as a base, supply its
-version to --from-version. Otherwise the SDK will scaffold a new CSV manifest.
+version to --from-version. Otherwise the SDK will scaffold a new CSV manifest.`,
 
-Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config.yaml`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			// The CSV generator assumes that the deploy and pkg directories are
-			// present at runtime, so this command must be run in a project's root.
-			projutil.MustInProjectRoot()
```

Comment:
Why it was removed? Is not required be in the Project root path to gen the CSV in the default path?

Reply:
This function does not really verify project root. All it does is check for `build/Dockerfile`, which is overly restrictive for no clear reason, when it comes to running this in non-SDK project layouts like kubebuilder, or SDK projects that move their `build/Dockerfile` to some other path.

And the purpose of this PR is to allow running `generate csv` in non-SDK layouts as much as possible.

Reply:
Hi @hasbro17, 

Currently, we check if is the root of the project by checking if the Dockerfile exist. I agree that after the integration it will change. We will probably change the implementation of this func to check if has the Project file or not in the current dir as it is done in kubebuilder. However, IMO we cannot remove this now. Note that:

- The command should return the error if it not be executed from the root dir. 
- The command should still work successfully whiteout the need to inform the "inputs". 

Then, we cannot the command be executed outside of the root project.

Reply:
@camilamacedo86 This change is not just for after the kubebuilder integration.
It's meant facilitate  running the `generate csv` command in a non-SDK project today.
E.g: https://github.com/knative/eventing-operator


> The command should return the error if it not be executed from the root dir

Why should the command not be run from the root directory?

> The command should still work successfully whiteout the need to inform the "inputs"

It will still do that without this check if you run it without any `inputs` flag in the root directory.

> we cannot the command be executed outside of the root project

I don't see any reason why not. The default still works. We're making the exception easier here i.e to run in a project that doesn't have `build/Dockerfile`. That's all our definition of root directory is anyway.

Reply:
What you said make completed sense for me. 👍 +1
I think you are 1000% right here.